### PR TITLE
Decompile some functions

### PR DESCRIPTION
**Pull request by octorock**

### CODE REVIEW

File: src/game.c [hunk 1]
```diff
@@ -1885,9 +1894,43 @@ void sub_08053758(void) {
...
+            gMenu.field_0xa--;
+        } else {
+            if (((gRoomTransition.frameCount & 1) == 0) && (gMenu.storyPanelIndex < 0x10)) {
+                tmp = ++gMenu.storyPanelIndex << 0x18;
+                gScreen.controls.alphaBlend = (tmp >> 0x10) | (0x10 - ((tmp) >> 0x19));
+            }
+        }
+        if (gMenu.transitionTimer == 0) {
+            gMenu.overlayType++;
+            SetFade(5, 8);
```
Comment (hatal175) [inline src/game.c:1911]: Use Fade constants

File: src/game.c [hunk 2]
```diff
@@ -1885,9 +1894,43 @@ void sub_08053758(void) {
...
+            gMenu.field_0xa--;
+        } else {
+            if (((gRoomTransition.frameCount & 1) == 0) && (gMenu.storyPanelIndex < 0x10)) {
+                tmp = ++gMenu.storyPanelIndex << 0x18;
+                gScreen.controls.alphaBlend = (tmp >> 0x10) | (0x10 - (tmp >> 0x19));
+            }
+        }
+        if (gMenu.transitionTimer == 0) {
+            gMenu.overlayType++;
+            SetFade(5, 1);
```
Comment (hatal175) [inline src/game.c:1930]: Use Fade constants

File: src/game.c [hunk 3]
```diff
@@ -2594,12 +2665,47 @@ void sub_080548E8(void) {
...
 
-ASM_FUNC("asm/non_matching/game/sub_08054920.inc", void sub_08054920())
+void sub_08054920(void) {
+    MessageInitialize();
+    sub_0806F38C();
+    if (gUnk_080C9CBC[gFuseInfo._3]._5[1] != 0) {
+        MenuFadeIn(10, gUI.field_0x3);
+    } else {
+        gUI.nextToLoad = 3;
+        SetFade(7, 0x10);
```
Comment (hatal175) [inline src/game.c:2675]: Use Fade constants

File: src/menu/kinstone_menu.c [hunk 4]
```diff
@@ -65,7 +65,16 @@ Subtask KinstoneMenu_Type3;
 Subtask KinstoneMenu_Type4;
 Subtask KinstoneMenu_Type5;
 
-ASM_FUNC("asm/non_matching/menu/kinstone_menu/sub_080A3B48.inc", s32 sub_080A3B48(void))
+u32 sub_080A3B48(void) {
```
Comment (hatal175) [inline src/menu/kinstone_menu.c:68]: Does this look better?
u32 sub_080A3B48(void) {
    u32 index;
    for (index = 0; index <= 0x12 && gSave.unk12B[index] != 0; index++) {
    }
    return index;
}

Not critical

File: src/playerUtils.c [hunk 5]
```diff
@@ -701,14 +715,103 @@ void ForceSetPlayerState(u32 framestate) {
...
+        case 0xc:
+        case 0x12:
+        case 0x15:
+        case 0x16:
+            return FALSE;
+    }
+    if ((gPlayerState.field_0x27[0] | gPlayerState.swim_state) != 0) {
+        return FALSE;
+    }
+    if (gPlayerState.floor_type == 1) {
```
Comment (hatal175) [inline src/playerUtils.c:739]: Use SurfaceType

File: src/playerUtils.c [hunk 6]
```diff
@@ -701,14 +715,103 @@ void ForceSetPlayerState(u32 framestate) {
...
+        if (gPlayerState.field_0x14 == 0) {
+            return FALSE;
+        }
+        if (CanDispEzloMessage()) {
+            return TRUE;
+        }
+        return FALSE;
+    }
+    if ((gPlayerState.flags & 0x4000) != 0) {
```
Comment (hatal175) [inline src/playerUtils.c:748]: Use PlayerState flags

File: src/playerUtils.c [hunk 7]
```diff
@@ -1181,15 +1301,61 @@ void sub_08079744(Entity* this) {
...
+}
 
 ASM_FUNC("asm/non_matching/playerUtils/sub_080797EC.inc", void sub_080797EC())
 
 ASM_FUNC("asm/non_matching/playerUtils/ResolvePlayerAnimation.inc", void ResolvePlayerAnimation())
 
-ASM_FUNC("asm/non_matching/playerUtils/sub_08079B24.inc", u32 sub_08079B24())
+bool32 sub_08079B24(void) {
+    if ((gPlayerEntity.action != 0xa) && (gPlayerEntity.health == 0)) {
+        gPlayerState.flags = gPlayerState.flags & 0xfeffffff;
```
Comment (hatal175) [inline src/playerUtils.c:1325]: Use ~PlayerState flags

File: src/playerUtils.c [hunk 8]
```diff
@@ -1181,15 +1301,61 @@ void sub_08079744(Entity* this) {
...
+        gPlayerEntity.knockbackDuration = gPlayerEntity.health;
+        if (gPlayerState.field_0x7 == 0) {
+            if (gPlayerState.swim_state != 0) {
+                RespawnPlayer();
+            } else {
+                if ((gPlayerState.field_0x14 == 0) && ((gPlayerState.flags & 0x1000) == 0)) {
```
Comment (hatal175) [inline src/playerUtils.c:1331]: Use PLayerState flags

File: src/playerUtils.c [hunk 9]
```diff
@@ -1181,15 +1301,61 @@ void sub_08079744(Entity* this) {
...
+                    if (gPlayerEntity.z.HALF.HI < 0) {
+                        if (0 < gPlayerEntity.zVelocity) {
+                            gPlayerEntity.zVelocity = gPlayerState.flags & 0x1000;
```
Comment (hatal175) [inline src/playerUtils.c:1334]: Isn't gPlayerState.flags & 0x1000 just 0 here?

File: src/playerUtils.c [hunk 10]
```diff
@@ -1181,15 +1301,61 @@ void sub_08079744(Entity* this) {
...
+                        if ((gPlayerState.jump_status & 0x41) == 0) {
+                            gPlayerState.jump_status = 0x41;
+                            gPlayerEntity.direction = 0xff;
+                            gPlayerState.field_0xd = 0xff;
+                            return TRUE;
+                        } else {
+                            return TRUE;
+                        }
+                    }
+                    if (((gPlayerState.flags & 2) == 0) && (sub_08079D48() == 0)) {
```
Comment (hatal175) [inline src/playerUtils.c:1345]: PlayerState flags

**Review by notyourav — APPROVED** [review]

**Pull request merged**